ncm-metaconfig/src/main/pan/components/metaconfig/schema.pan

```diff
@@ -118,9 +130,26 @@ type ${project.artifactId}_config =  {
     'contents' : ${project.artifactId}_extension
     @{Predefined conversions from EDG::WP4::CCM::TextRender}
     'convert' ? ${project.artifactId}_textrender_convert
+    @{Actions (i.e. named registered commands) to run on pre, validation and/or post step.
+      These are independent of daemons and are executed at time of processing each service.}
+    'actions' ? ${project.artifactId}_actions
 } = dict();
 
 type ${project.artifactId}_component = {
     include structure_component
     'services' : ${project.artifactId}_config{} with valid_absolute_file_paths(SELF)
+    @{Command registry for allowed actions, keys should be used as action value}
+    'commands' ? string{}
```

Comment:
I think we need to say something about how commands are executed. i.e. (depending on discussion below): The commands will be executed via CAF::Process without shell.